contracts/ForkableRealityETH_ERC20.sol

```diff
@@ -2,977 +2,173 @@
-    /// @param template_id The ID number of the template the question will use
-    /// @param question A string containing the parameters that will be passed into the template to make the question
-    /// @param arbitrator The arbitration contract that will have the final word on the answer if there is a dispute
-    /// @param timeout How long the contract should wait after the answer is changed before finalizing on that answer
-    /// @param opening_ts If set, the earliest time it should be possible to answer the question.
-    /// @param nonce A user-specified nonce used in the question ID. Change it to repeat a question.
-    /// @return The ID of the newly-created question, created deterministically.
-    function askQuestion(
-        uint256 template_id,
-        string memory question,
-        address arbitrator,
-        uint32 timeout,
-        uint32 opening_ts,
-        uint256 nonce
-    )
-        public
-        returns (
-            // stateNotCreated is enforced by the internal _askQuestion
-            bytes32
-        )
-    {
-        require(templates[template_id] > 0, "template must exist");
-
-        bytes32 content_hash = keccak256(
-            abi.encodePacked(template_id, opening_ts, question)
-        );
-        bytes32 question_id = keccak256(
-            abi.encodePacked(
-                content_hash,
-                arbitrator,
-                timeout,
-                msg.sender,
-                nonce
+    // This arbitrator is assigned automatically when importing questions
+    address public l1ForkArbitrator;
+
+    uint256 constant UPGRADE_TEMPLATE_ID = 1048576;
+
+    function initialize(
+        address _forkmanager,
+        address _parentContract,
+        address _token,
+        bytes32 _questionIdWeForkedOver
+    ) public initializer {
+        // We do this with a new contract instead of a proxy as it's pretty tiny
+        // TODO: Should we use a proxy pattern like elsewhere?
+        l1ForkArbitrator = address(
+            new L1ForkArbitrator(
+                address(this),
+                address(_forkmanager),
+                address(_token)
             )
         );
 
-        _askQuestion(
-            question_id,
-            content_hash,
-            arbitrator,
-            timeout,
-            opening_ts
-        );
-        emit LogNewQuestion(
-            question_id,
-            msg.sender,
-            template_id,
-            question,
-            content_hash,
-            arbitrator,
-            timeout,
-            opening_ts,
-            nonce,
-            block.timestamp
-        );
-
-        return question_id;
-    }
-
-    function _deductTokensOrRevert(uint256 tokens) internal {
-        if (tokens == 0) {
-            return;
-        }
+        ForkableStructure.initialize(_forkmanager, _parentContract);
 
-        uint256 bal = balanceOf[msg.sender];
+        _createInitialTemplates();
+        token = IERC20(_token);
```

Comment:
you can require the function to receive an IERC20 parameter, then you don't have to convert it here.

Reply:
Generally our fork creation stuff takes and returns addresses, it would be weird to make this an exception.